src/flutter/code-tools.ts

```diff
@@ -0,0 +1,284 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as Sentry from '@sentry/node';
+// @ts-ignore - clack is ESM and TS complains about that. It works though
+import * as clack from '@clack/prompts';
+import chalk from 'chalk';
+import {
+  sentryImport,
+  pubspecOptions,
+  sentryProperties,
+  initSnippet,
+} from './templates';
+import { featureSelectionPrompt } from '../utils/clack-utils';
+
+/**
+ * Recursively finds a file per name in subfolders.
+ * @param dir - The directory to start searching.
+ * @param name - The name of the file including path extension.
+ * @returns The path to the main.dart file or null if not found.
+ */
+export function findFile(dir: string, name: string): string | null {
+  const files: string[] = fs.readdirSync(dir);
+
+  for (const file of files) {
+    const fullPath: string = path.join(dir, file);
+    const stats: fs.Stats = fs.statSync(fullPath);
+
+    if (stats.isDirectory()) {
+      const result: string | null = findFile(fullPath, name);
+      if (result) {
+        return result;
+      }
+    } else if (file === name) {
+      return fullPath;
+    }
+  }
+
+  return null;
+}
+
+export function patchPubspec(
+  pubspecFile: string | null,
+  sentryDartFlutterVersion: string,
+  sentryDartPluginVersion: string,
+  project: string,
+  org: string,
+): boolean {
+  try {
+    if (!pubspecFile) {
+      throw new Error('pubspec.yaml is not provided or invalid.');
+    }
+
+    let pubspecContent = fs.readFileSync(pubspecFile, 'utf8');
+
+    if (!pubspecContent.includes('sentry_flutter:')) {
+      const dependenciesIndex = getDependenciesLocation(pubspecContent);
+
+      pubspecContent =
+        pubspecContent.slice(0, dependenciesIndex) +
+        `  sentry_flutter: ${sentryDartFlutterVersion}\n` +
+        pubspecContent.slice(dependenciesIndex);
+
+      clack.log.success(
+        chalk.greenBright(
```

Comment:
nit: since we use `chalk.greenBright` a lot we could maybe extract it into a `function logSuccess` but up to you

Reply:
Using it the same as in other wizards.